internal/gatewayapi/translator.go

```diff
@@ -18,6 +18,13 @@ const (
 	KindHTTPRoute = "HTTPRoute"
 	KindService   = "Service"
 	KindSecret    = "Secret"
+
+	// OwningGatewayNameLabel is the owner reference label used for managed infra.
+	// The value should be the name of the Gateway used to CRUD the Service.
+	OwningGatewayNameLabel = "gateway.envoyproxy.io/owning-gateway-name"
+	// OwningGatewayNamespaceLabel is the owner reference label used for managed infra.
+	// The value should be the namespace of the Gateway used to CRUD the Service.
+	OwningGatewayNamespaceLabel = "gateway.envoyproxy.io/owning-gateway-namespace"
```

Comment:
Doesn't the infra correspond to a GatewayClass, which could have >1 Gateways? I would think these labels should identify the GatewayClass.

Reply:
@skriss since a Gateway supports multiple addresses, each Gateway corresponds to a Service. Therefore, a Service name is constructed as `envoy-${GW_NAME}-${GW_NS}`, where the variables are derived from these labels.